src/kmad.cpp

```diff
@@ -169,26 +167,22 @@ int main(int argc, char *argv[]) {
 
     // Combine sequence and feature settings
     //
-    auto sequence_data_plain = seq_data::process_fasta_data(
-        fasta_data, f_set, gapped
-    );
+    fasta_data = fasta::get_conf_data(fasta_data, f_set, gapped);
 
     // Perform the alignment
 
     // TODO: Maybe we can use a design pattern here because the type of
     // alignment depends on whether or not we will refine.
     std::vector<fasta::SequenceList> alignment;
     if (!refine) {
-      alignment = msa::run_msa(sequence_data_plain,
-                               f_set, gap_open_pen,
-                               gap_ext_pen, end_pen, domain_modifier,
-                               motif_modifier, ptm_modifier, strct_modifier,
-                               codon_length,
-                               one_round, sbst_mat, first_gapped, optimize,
-                               fade_out, no_feat);
+      alignment = msa::run_msa(
+                      fasta_data, f_set, gap_open_pen, gap_ext_pen, end_pen,
+                      domain_modifier, motif_modifier, ptm_modifier,
+                      strct_modifier, codon_length, one_round, sbst_mat,
+                      first_gapped, optimize, fade_out, no_feat);
     } else {
       bool gapped = true;
-      seq_data::SequenceData sequence_data_alignment = seq_data::process_fasta_data(
+      fasta::FastaData fasta_data_alignment = fasta::get_conf_data(
```

Comment:
It needs to be called again with the gapped variable set to true (only in the refine mode though), because we need both with & eithout gaps -> and feature positions are different in these, so that's why you need to reassign the features to the residues in the alignment.